lib/router/website_route_information_parser.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:collection/collection.dart';
+import 'package:festival_flutterturkiye_org/core/model/speaker.dart';
+import 'package:festival_flutterturkiye_org/router/website_configuration.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+const unknownPageSegmentName = '404';
+
+final eventPathInformation = PathInformation(
+  pathSegmentName: 'event',
+  focusNodeIndex: 0,
+);
+
+final programPathInformation = PathInformation(
+  pathSegmentName: 'program',
+  focusNodeIndex: 2,
+  queryParameter: 'speaker',
+);
+
+final sponsorPathInformation = PathInformation(
+  pathSegmentName: 'sponsors',
+  focusNodeIndex: 3,
+);
+
+final faqPathInformation = PathInformation(
+  pathSegmentName: 'faq',
+  focusNodeIndex: 4,
+);
+
+final contactUsPathInformation = PathInformation(
+  pathSegmentName: 'contact-us',
+  focusNodeIndex: 5,
+);
+
+class PathInformation {
+  PathInformation({
+    required this.pathSegmentName,
+    required this.focusNodeIndex,
+    this.queryParameter,
+  });
+
+  final String pathSegmentName;
+  final int focusNodeIndex;
+  final String? queryParameter;
+}
+
+List<PathInformation> availablePathInformationList = [
+  eventPathInformation,
+  programPathInformation,
+  sponsorPathInformation,
+  faqPathInformation,
+  contactUsPathInformation,
+];
+
+List<String> get availablePathSegmentNames =>
+    availablePathInformationList.map((e) => e.pathSegmentName).toList();
+
+class WebsiteRouteInformationParser
+    extends RouteInformationParser<WebsiteConfiguration> {
+  WebsiteRouteInformationParser({required this.speakers});
+
+  final List<Speaker> speakers;
+
+  @override
+  Future<WebsiteConfiguration> parseRouteInformation(
+    RouteInformation routeInformation,
+  ) async {
+    final uri = Uri.parse(routeInformation.location!);
```

Comment:
probably it would be safer to call `Uri.tryParse` since `routeInformation.location` is nullable. And if null to return `WebsiteConfiguration.unknown`. Otherwise this will throw if it is null. Or is there something I'm missing?

Reply:
Just realized that `location` field can be null but I never had a case where it was null. Fixed b24b3195e7a33489e7b3c951559d6b2bee04e502